xla/literal.cc

```diff
@@ -1844,6 +1844,16 @@ bool LiteralBase::Piece::EqualElements(const LiteralBase::Piece& other) const {
     CHECK(LayoutUtil::IsDenseArray(subshape()))
         << __func__ << " is only supported for dense arrays: " << subshape();
     CHECK_EQ(size_bytes_dense(), other.size_bytes_dense());
+    auto one_array = buffer();
+    auto two_array = other.buffer();
+    if (primitive_util::Is4BitType(subshape().element_type())) {
+      for (int64_t i = 0; i < size_bytes_dense(); ++i) {
+        if ((*(one_array + i) & uint8_t{0xf}) !=
```

Comment:
Use notation `one_array[i]` instead of `*(one_array + i)`.

Reply:
Done